pkg/provider/onepassword/onepassword.go

```diff
@@ -0,0 +1,364 @@
+
+	// handle files
+	if item.Category == documentCategory {
+		for _, file := range item.Files {
+			if ref.Property != "" && file.Name != ref.Property {
+				continue
+			}
+			contents, err := provider.client.GetFileContent(file)
+			if err != nil {
+				return nil, err
+			}
+			secretData[file.Name] = contents
+		}
+
+		return secretData, nil
+	}
+
+	// handle fields
+	for _, field := range item.Fields {
+		if ref.Property != "" && field.Label != ref.Property {
+			continue
+		}
+		if length := countFieldsWithLabel(field.Label, item.Fields); length != 1 {
+			return nil, fmt.Errorf(errExpectedOneField, fmt.Errorf("'%s' in '%s', got %d", field.Label, item.Title, length))
+		}
+
+		// caution: do not use client.GetValue here because it has undesirable behavior on keys with a dot in them
+		secretData[field.Label] = []byte(field.Value)
+	}
+
+	return secretData, nil
+}
+
+// GetAllSecrets syncs multiple 1Password Items into a single Kubernetes Secret, for dataFrom.find.
+func (provider *ProviderOnePassword) GetAllSecrets(ctx context.Context, ref esv1beta1.ExternalSecretFind) (map[string][]byte, error) {
+	secretData := make(map[string][]byte)
+	sortedVaults := sortVaults(provider.vaults)
+	for _, vaultName := range sortedVaults {
+		vaults, err := provider.client.GetVaultsByTitle(vaultName)
+		if err != nil {
+			return nil, fmt.Errorf(errGetVault, err)
+		}
+		if len(vaults) != 1 {
+			return nil, fmt.Errorf(errExpectedOneVault, fmt.Errorf("'%s', got %d", vaultName, len(vaults)))
+		}
+
+		items, err := provider.client.GetItems(vaults[0].ID)
+		if err != nil {
+			return nil, fmt.Errorf(errGetItem, err)
+		}
+		for _, item := range items {
+			if ref.Path != nil && *ref.Path != item.Title {
+				continue
+			}
+
+			// handle files
+			if item.Category == documentCategory {
+				for _, file := range item.Files {
+					matcher, err := find.New(*ref.Name)
+					if err != nil {
+						return nil, err
+					}
+					if !matcher.MatchName(file.Name) {
+						continue
+					}
+					if _, ok := secretData[file.Name]; !ok {
+						contents, err := provider.client.GetFileContent(file)
+						if err != nil {
+							return nil, err
+						}
+						secretData[file.Name] = contents
+					}
+				}
+
+				continue
+			}
+
+			// handle fields
+			i, err := provider.client.GetItem(item.ID, item.Vault.ID)
+			if err != nil {
+				return nil, fmt.Errorf(errGetItem, err)
+			}
+			item = *i
+			for _, field := range item.Fields {
+				if length := countFieldsWithLabel(field.Label, item.Fields); length != 1 {
+					return nil, fmt.Errorf(errExpectedOneField, fmt.Errorf("'%s' in '%s', got %d", field.Label, item.Title, length))
+				}
+				matcher, err := find.New(*ref.Name)
+				if err != nil {
+					return nil, err
+				}
+				if !matcher.MatchName(field.Label) {
+					continue
+				}
+				if _, ok := secretData[field.Label]; !ok {
+					secretData[field.Label] = []byte(field.Value)
+				}
+			}
+		}
+	}
+
+	return secretData, nil
+}
```

Comment:
If 1password does not support Find By Tags, can you implement a  Not Supported error if `find.tags` is filled?

Reply:
If it does support, but it is not implemented, can you add a Not Implemented error? :D

Reply:
Added not implemented error and relevant tests.